### ISSUE TITLE
Adding support for linux

### ISSUE THREAD

**Issue by Shikhar-SRJ** [AndreMiras/pycaw#46]
Although, it is clearly stated that this is a library for windows only.  Yet we can't deny the fact that there are just 45% windows users and the rest are 29% macOS and about 25% Linux users.   
Therefore, I suggest adding compatibility with Linux as well. This will make this library a bit more standard and not limited to a specific os.

**Comment by TurboAnonym** [comment]
I think its easier to have a seperate package for linux, since it is all highly specific. When you are using pycaw u need to now how the windows audio session api, device api and com in generall works. Pycaw is mostly not fancy and abstract away (besides the magic module and the utils module). Pycaw mostly provides (when implemented) a way to use the windows core audio api as you would in C++ or so on.

But you could create a wrapper around pycaw and some linux audio python library to cross plattform control audio.

**Comment by Shikhar-SRJ** [comment]
I agree with your opinion of using a separate package for Linux. 

Actually, what I am pointing at is to make the library independent of the operating system it runs on. And I am very curious to know if it is feasible and if yes, then how?

My apologies if this question seems stupid, but why is this compatibility not yet added? Is it because of the increased complexity, or are there any other reasons?

**Comment by AndreMiras** [comment]
Let me give a bit of back story first.
First of all I'm a long time 100% Linux user.
I've been running Linux as my prior operating system for 15 years.
This means I need to install a Windows VM everytime I want to maintain pycaw by the way.
Then I built pycaw for a friend who needed it since it was nothing like that for Windows.

There're currently no abstraction library or standard that I'm aware of that works cross operating system.
The way theses things usually work is someone build a low level wrapper for an OS like I did for pycaw, then someone else probably also built something similar for another OS.
Then eventually another dude that wants to use all 3 in an abstract way will use all of them to build an abstract one.
I'm not this dude because I built pycaw for a friend, but don't need it or use it
